test/libsolidity/util/compiler/InternalCompiler.cpp

```diff
@@ -0,0 +1,180 @@
+#include <test/libsolidity/util/compiler/InternalCompiler.h>
+
+using namespace solidity;
+using namespace solidity::evmasm;
+using namespace solidity::frontend::test;
+
+CompilerOutput InternalCompiler::compile(CompilerInput const& _input)
+{
+    configure(_input);
+
+    auto [contracts, errors] = internalCompile(_input);
+    bool success = m_stack.compilationSuccessful();
+
+    return CompilerOutput{
+        contracts,
+        success,
+        errors,
+        formatErrorInformation()
+    };
+}
+
+void InternalCompiler::configure(CompilerInput const& _input)
+{
+    m_stack.reset();
+
+    m_stack.setSources(_input.sourceCode);
+    m_stack.setLibraries(_input.libraryAddresses);
+
+    if (_input.evmVersion.has_value())
+        m_stack.setEVMVersion(_input.evmVersion.value());
+    if (_input.optimise.has_value())
+        m_stack.setOptimiserSettings(_input.optimise.value());
+    if (_input.optimiserSettings.has_value())
+        m_stack.setOptimiserSettings(_input.optimiserSettings.value());
+    if (_input.revertStrings.has_value())
+        m_stack.setRevertStringBehaviour(_input.revertStrings.value());
+    if (_input.metadataFormat.has_value())
+    {
+        CompilerStack::MetadataFormat metadata = CompilerStack::MetadataFormat::NoMetadata;
```

Comment:
can't you use the already existing enums here? same for the hash.

Reply:
There was something blocking me from using the existing ones in the first place. But that blocker seems to be gone, so happy to remove them :)